Analyzer/src/AnalyzeLepTrigger.cc

```diff
@@ -109,356 +100,144 @@ void AnalyzeLepTrigger::Loop(NTupleReader& tr, double weight, int maxevents, boo
-        bool passElectronTriggers   = passTriggerGeneral( myElectronTriggers, TriggerNames, TriggerPass );
+        bool passMuonTriggers       = tr.getVar<bool>("passTriggerMuon");
+        bool passElectronTriggers   = tr.getVar<bool>("passTriggerElectron");
         
-        //Define offline baseline for one lepton analysis
-        bool passOfflineBaseline    = ( JetID && passMadHT && NGoodBJets_pt30 >= 1 && Mbl > 50 && Mbl < 250 && HT_trigger_pt30 > 300 );
-
         bool pass6JetCut            = ( NGoodJets_pt30 >= 6 );
         bool pass5JetCut            = ( NGoodJets_pt30 >= 5 );
         bool pass4JetCut            = ( NGoodJets_pt30 >= 4 );
         bool pass3JetCut            = ( NGoodJets_pt30 >= 3 );
         bool pass2JetCut            = ( NGoodJets_pt30 >= 2 );
         bool pass1JetCut            = ( NGoodJets_pt30 >= 1 ); //A little redundant because we require one good b jet but for symmetry reasons
 
-        bool atLeastOneGoodEl       = ( std::any_of( GoodElectrons.begin(), GoodElectrons.end(), [] ( bool boolEl ) { return boolEl; } ) );
-        bool atLeastOneGoodMu       = ( std::any_of( GoodMuons.begin(), GoodMuons.end(), [] ( bool boolMu ) { return boolMu; } ) );
-
-
-        //------------------------------------------------------
-        //-- Do the Electron Trigger Efficiency on the Single Muon Dataset
-        //------------------------------------------------------
+        //-----------------------------------------------------------------------
+        //-- Do the Electron Trigger Efficiency on the Single Muon Dataset and MC
+        //-----------------------------------------------------------------------
         
-        if( filetag == "2016_Data_SingleMuon" || filetag == "2017_Data_SingleMuon" ) {
-            
-            if( !passOfflineBaseline || !atLeastOneGoodMu ) continue;
+        if( (filetag.find("SingleMuon") != std::string::npos || runtype == "MC") ) {
 
-            bool foundMuonPt35      = false;
-            bool foundMuonPt40      = false;
+            bool atLeastOneGoodMu = ( NGoodMuons >= 1 );
             
-            //Require a good muon in the single muon dataset
-            for( unsigned int itMu = 0; itMu < Muons.size(); ++itMu ) {
-                if( !GoodMuons.at( itMu ) ) continue; 
-
-                TLorentzVector myMuon = Muons.at( itMu );
-
-                if( myMuon.Pt() >= 35 && std::fabs( myMuon.Eta() ) < 2.4 )       foundMuonPt35 = true;
-                if( myMuon.Pt() >= 40 && std::fabs( myMuon.Eta() ) < 2.4 )       foundMuonPt40 = true;
-            }
-
-            //Look at the first good electron
-            int myGoodElectronIndex     = -1;
-            for( unsigned int itEl = 0; itEl < Electrons.size(); ++itEl ) {
-                if( !GoodElectrons.at( itEl ) ) continue;
-
-                myGoodElectronIndex = itEl;
-                break;
-            }
+            if ( atLeastOneGoodMu ) {
```

Comment:
is atLeastOneGoodMu used any where else? If not then there is no need to make this variable, and this line should be:
if( NGoodMuons >= 1)

Reply:
Nope, only used there. Simplified.